### PR TITLE
feat: add R tree-sitter symbol extractor

### PR DESCRIPTION
**Pull request by justinjdev**
## Summary
- Add tree-sitter-based symbol extraction for R language files (`.r`, `.R`)
- Extract function definitions via `<-` and `=` assignment
- Extract S4 class definitions (`setClass`, `setRefClass`)
- Extract R6 class definitions (`R6Class`)
- Extract function parameters including `...` (dots/varargs)

## Implementation
- `tree-sitter-r = "1.2"` crate (compatible with tree-sitter 0.24)
- 4 tree-sitter query patterns with post_process hook filtering for class definitions
- 11 unit tests covering functions, classes, parameters, edge cases

## Test plan
- [x] `cargo check` passes
- [x] `cargo test --lib` passes (11 new R tests + no regressions)
- [x] Adversarial review completed (balrog)
- [x] Uppercase `.R` extension verified working

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added R language support for source code symbol extraction, enabling recognition of functions (with parameters) and classes (S4 and R6 syntaxes).
  * File extensions `.r` and `.R` now recognized.

* **Documentation**
  * Updated supported languages list to include R.

### CODE REVIEW

File: src/symbols/walker.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ pub fn all_extensions() -> Vec<&'static str> {
         "scala", "sc",                       // Scala
         "ex", "exs",                         // Elixir
         "zig",                               // Zig
-        "sh", "bash",                        // Bash/Shell
+        "r", "R",                            // R
```
[the remark this answers is not in the document]
Comment (justinjdev) [reply src/symbols/walker.rs]: Fixed in 286f264. Added `assert!(exts.contains(&"r"));` and `assert!(exts.contains(&"R"));` to `test_all_extensions`.
